### PR TITLE
fillers/vm/vm_arithmetic: Port vmArithmetic fillers

### PR DESCRIPTION
**Pull request by spencer-tb**
PR for the ported tests of https://github.com/ethereum/tests/tree/develop/src/GeneralStateTestsFiller/VMTests/vmArithmeticTest.

**For the fixtures generated we have:**
- add_opcode.json
- sub_opcode.json
- arith.json
- div_by_zero.json
- exp_opcode.json
- exp_power_2.json
- exp_power_256_of_256.json
- exp_power_256.json
- addmod_opcode.json
- mod_opcode.json
- mulmod_opcode.json
- div_opcode.json
- mul_opcode.json
- sdiv_opcode.json
- not_opcode.json

### CODE REVIEW

**Review by marioevz — COMMENTED** [review]
Some comments, I have not finished reviewing yet :)

File: src/ethereum_test_tools/common/helpers.py [hunk 1]
```diff
@@ -95,5 +95,5 @@ def to_hash(input: int | str) -> str:
         # Convert to int
         input = int(input, 0)
     if type(input) is int:
-        return "0x" + input.to_bytes(32, "big").hex()
+        return "0x" + input.to_bytes(32, "big", signed=True).hex()
```
[the remark this answers is not in the document]
Comment (spencer-tb) [reply src/ethereum_test_tools/common/helpers.py:98]: Oh nice find. Will add that fix & addition to doc string

File: fillers/vm/vm_arithmetic/mul_div.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        + Op.SDIV
+        + Op.PUSH1(0x20)
+        + Op.SSTORE
+        + Op.STOP
+    )
+
+    pre = {
+        to_address(0x100): Account(
+            balance=0x0BA1A9CE0BA1A9CE,
+            code=Yul(
```
[the remark this answers is not in the document]
Comment (spencer-tb) [reply fillers/vm/vm_arithmetic/mul_div.py:215]: Sure sounds good!

File: fillers/vm/vm_arithmetic/exp.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        ("2, 0x0100000000000f", 0x00),
+        ("2, 15", 0x8000),
+    ]
+
+    for i, (b_e, result) in enumerate(base_exp_result):
+        address = to_address(0x100 + i)
+        yul_code = Template(
+            """
+            {
+                sstore(${address}, exp(${b_e}))
```
[the remark this answers is not in the document]
Comment (spencer-tb) [reply fillers/vm/vm_arithmetic/exp.py]: Ahh yeah, I kept that that there for testing, I will remove it!

**Comment by spencer-tb** [comment]
CIFO `ethereum/tests` port issue: https://github.com/ethereum/execution-spec-tests/issues/391

**Pull request closed without merging**